addons/udes_security/models/ir_actions.py

```diff
@@ -0,0 +1,16 @@
+from collections import defaultdict
+
+from odoo import api, models, tools
+
+
+class IrActions(models.Model):
+    _inherit = "ir.actions.actions"
+
+    @api.model
+    @tools.ormcache("frozenset(self.env.user.groups_id.ids)", "model_name")
```

Comment:
This could cause issue. It may not get updated when user permissions are changed from frontend.
So test the following scenario
* Start the server
* Change the permissions
* Test the permissions
* Change the permission and then test the permissions

Reply:
This was taken from the original get_bindings method in `base/ir/ir_actions.py` so kept it in for safety, can try with/without to see if it makes a difference though. This permission is a boolean flag on the user so should be fine either way but will look into it. :)

Reply:
All good with and without this being set, but since I'm not using `groups_id` there is no point in keeping it, so have removed.